src/bundler.js

```diff
@@ -374,47 +401,51 @@ function processCssBundle(options, cssBundle, bundleDir, cssFiles, bundleName, c
 
     var allCssArr = [], allMinCssArr = [], index = 0, pending = 0;
     var whenDone = function () {
-        var allCss = concat.files({
+
+        allCssArr.forEach(function(cssFile) {
+            bundleStatsCollector.AddDebugFile(cssBundle, cssFile.path);
+        });
+
+        concat.files({
                 files: allCssArr,
                 fileType: file.type.CSS,
                 sourceMap: options.sourcemaps
-            }),
-            allMinCss = concat.files({
-                files: allMinCssArr,
-                fileType: file.type.CSS,
-                sourceMap: options.sourcemaps
-            });
+            })
+            .then(function(allCss) {
 
-        var afterBundle = function () {
+                return file.write(allCss.code, allCss.map, file.type.CSS, bundleName, options.siterootdirectory);
 
-            if(urlVersioning) {
-                allMinCss.code = urlVersioning.VersionUrls(allMinCss.code);
-            }
+            })
+            .then(function() {
+
+                return concat.files({
+                    files: allMinCssArr,
+                    fileType: file.type.CSS,
+                    sourceMap: options.sourcemaps
+                });
 
-            cssValidator.validate(cssBundle, allMinCss.code, function(err) {

```

Comment:
In converting this to a chain of promises, I updated our CSS validator step to use a promise as well, rather than a callback.